modules/tinymce/src/core/test/ts/browser/util/XhrTest.ts

```diff
@@ -6,8 +6,8 @@ import XHR, { XHRSettings } from 'tinymce/core/api/util/XHR';
 UnitTest.asynctest('browser.tinymce.core.util.XhrTest', function (success, failure) {
   const suite = LegacyUnit.createSuite();
 
-  type XHRTest = XMLHttpRequest & { test: number };
-  type XHRSettingsTest = XHRSettings & { test: number };
+  type XHRTest = XMLHttpRequest & { test?: number };
```

Comment:
This looks to be incorrect... as this test always passes test (as per the beforeSend hook), so it shouldn't be optional.

Reply:
The issue is that in the callback we specify that the event we receive from the XHR is of type `{ xhr: XHRTest; settings: XHRSettingsTest }`. While we do always add the test field in the beforeSend hook, the event sent through by the editor never has the test field.